plugins/UsersManager/API.php

```diff
@@ -264,6 +267,64 @@ private function getDefaultUserPreference($preferenceName, $login)
         }
     }
 
+    /**
+     * Returns all users with their role for $idSite.
+     *
+     * @param int $idSite
+     * @param int|null $limit
+     * @param int|null $offset
+     * @param string|null $filter_search text to search for in the user's login, email and alias (if any)
+     * @param string|null $filter_access only select users with this access to $idSite. can be 'noaccess', 'some', 'view', 'admin', 'superuser'
+     *                                   Filtering by 'superuser' is only allowed for other superusers.
+     * @return array
+     */
+    public function getUsersPlusRole($idSite, $limit = null, $offset = 0, $filter_search = null, $filter_access = null)
+    {
+        if (!$this->isUserHasAdminAccessTo($idSite)) {
+            // if the user is not an admin to $idSite, they can only see their own user
+            if ($offset > 1) {
+                return [
+                    'total' => 1,
+                    'results' => [],
+                ];
+            }
+
+            $user = $this->model->getUser($this->access->getLogin());
+            $user['role'] = $this->access->getRoleForSite($idSite);
+            $user['capabilities'] = $this->access->getCapabilitiesForSite($idSite);
+            $users = [$user];
```

Comment:
Would this return password and other information which we might not want to expose even if it is the users' own information?

Reply:
token should also not be returned.

Reply:
`removeUserInfoForNonSuperUsers()` and `enrichUsers()` are both called below. I think password was always returned in some cases, but can unset it in enrichUser (where token_auth is unset).